controllers/imagepolicy_controller.go

```diff
@@ -203,35 +177,20 @@ func (r *ImagePolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	if err != nil {
-		imagev1.SetImagePolicyReadiness(
-			&pol,
-			metav1.ConditionFalse,
-			meta.ReconciliationFailedReason,
-			err.Error(),
-		)
-		if err := r.patchStatus(ctx, req, pol.Status); err != nil {
-			return ctrl.Result{Requeue: true}, err
-		}
-		r.event(ctx, pol, events.EventSeverityError, err.Error())
-		return ctrl.Result{}, err
-	}
-
-	if latest == "" {
-		msg := fmt.Sprintf("Cannot determine latest tag for policy: %s", err.Error())
+	if err != nil || latest == "" {
 		pol.Status.LatestImage = ""
-		imagev1.SetImagePolicyReadiness(
-			&pol,
-			metav1.ConditionFalse,
-			meta.ReconciliationFailedReason,
-			msg,
-		)
-
-		if err := r.patchStatus(ctx, req, pol.Status); err != nil {
-			return ctrl.Result{}, err
+		if err == nil {
+			err = fmt.Errorf("Cannot determine latest tag for policy")
+		} else {
+			err = fmt.Errorf("Cannot determine latest tag for policy: %w", err)
 		}
-		r.event(ctx, pol, events.EventSeverityError, msg)
-		return ctrl.Result{}, nil
+		res, recErr := recordError(err, meta.ReconciliationFailedReason)
+		if recErr != nil {
+			// log the actual error since we are returning the error related to patching status
+			log.Error(err, "")
+			return res, recErr
+		}
```

Comment:
Wondering if it would maybe be better to do:

```suggestion
		res, recErr := recordError(err, meta.ReconciliationFailedReason)
		if recErr != nil {
			log.Error(retErr, "failed to record reconciliation failure")
		}
		return ctrl.Result{}, err
```

Reply:
The `log.Error(err, "")` is actually meant to record the actual error. `recErr` is the error returned if there was an error while actually recording the error. Maybe this could provide more context: https://github.com/fluxcd/image-reflector-controller/commit/046c29e69dd09af0dba1ba48bd3634c628160621#r66637836